src/plugins/shared-variables/graph/plotted-variables-adornment/plotted-variables-adornment-component.tsx

```diff
@@ -215,50 +208,83 @@ export const PlottedVariablesAdornmentComponent = observer(function PlottedVaria
   useEffect(function refreshAxisChange() {
-    return mstAutorun(() => {
-      // We observe changes to the axis domains within the autorun by extracting them from the axes below.
-      // We do this instead of including domains in the useEffect dependency array to prevent domain changes
-      // from triggering a reinstall of the autorun.
-      if (xAxis && yAxis) {
-        const { domain: xDomain } = xAxis; // eslint-disable-line unused-imports/no-unused-vars
-        const { domain: yDomain } = yAxis; // eslint-disable-line unused-imports/no-unused-vars
-      }
-      // Trigger an autorun if any inputs or the expression of y change, or if the x variable changes
-      Array.from(model.plottedVariables.values()).forEach(plottedVariables => {
-        plottedVariables.yVariable?.computedValueIncludingMessageAndError; // eslint-disable-line no-unused-expressions
-        plottedVariables.xVariable; // eslint-disable-line no-unused-expressions
-      });
-      refreshValues();
-    }, { name: "PlottedVariablesAdornmentComponent.refreshAxisChange" }, model);
-  }, [dataConfig, model, plotWidth, plotHeight, sharedVariables, xAxis, yAxis, refreshValues]);
+    return mstReaction<string,boolean>(
+      () => {
+        if (xAxis && yAxis) {
+          const { domain: xDomain } = xAxis;
+          const { domain: yDomain } = yAxis;
+          // Return a primitive value (string, not array) so that equality will work
+          // and effect will only be called when these numbers actually change.
+          return `${xDomain[0]}, ${xDomain[1]}, ${yDomain[0]}, ${yDomain[1]}`;
+        }
+        return '';
+      },
+      (limits) => {
+        refreshValues();
+      },
+      { fireImmediately: true, name: "PlottedVariablesAdornmentComponent.refreshAxisChange" },
+      model);
+  }, [model, plotWidth, plotHeight, xAxis, yAxis, refreshValues, controller]);
+
+  // Refresh display when variable values change
+  useEffect(function udpateOnValuesChange() {
+    return mstReaction(
+      () => {
+        let hasAnyVars = false;
+        const varValues = Array.from(model.plottedVariables.values()).map(plottedVariables => {
+          if (plottedVariables.xVariable || plottedVariables.yVariable) {
+            hasAnyVars = true;
+          }
+          return {
+            x: plottedVariables.xVariable,
+            y: plottedVariables.yVariable?.computedValueIncludingMessageAndError
+          };
+        });
+        return {hasAnyVars, varValues};
+      },
+      (args: {hasAnyVars: boolean, varValues: any}) => {
+        if (args.hasAnyVars) {
+          if (settings.scalePlotOnValueChange && !graphModel.lockAxes) {
+            controller?.autoscaleAllAxes();
+          }
+          refreshValues();
+        }
+      },
+      { name: "PlottedVariablesAdornmentComponent.rescaleOnValuesChange" },
+      model);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [controller, model]);
```

Comment:
Is disabling the warning necessary? Seems like you ought to be able to do:
```suggestion
  }, [controller, graphModel, model, refreshValues, settings]);
```
The only one of these that isn't necessarily referentially stable is `refreshValues`, but if that has changed not listing it as a dependency would result in calling the stale version.